### ISSUE TITLE
Adding a network interface kills a running source and leaves a zombie

### ISSUE THREAD

**Issue by kinnarr** [metricq/metricq-source-sysinfo#6]
Adding a new network interface while the source is running, raises an error and leaves a running but unresponsive source.

```
Oct 07 10:24:32 igel metricq-source-sysinfo[753]: 2022-10-07 10:24:32,960 [ERROR   ] [metricq.agent       ] Exception in event loop: Task exception was never retrieved
Oct 07 10:24:33 igel metricq-source-sysinfo[753]: 2022-10-07 10:24:32,963 [ERROR   ] [metricq.agent       ] Future: <Task finished name='Task-35' coro=<IntervalSource.task() done, de
Oct 07 10:24:34 igel metricq-source-sysinfo[753]: 2022-10-07 10:24:33,181 [ERROR   ] [metricq.agent       ] Agent SysinfoSource encountered an unhandled exception
Oct 07 10:24:34 igel metricq-source-sysinfo[753]: Traceback (most recent call last):
Oct 07 10:24:34 igel metricq-source-sysinfo[753]:   File "/home/service/envs/sysinfo/lib/python3.9/site-packages/metricq/interval_source.py", line 131, in task
Oct 07 10:24:34 igel metricq-source-sysinfo[753]:     await self.update()
Oct 07 10:24:34 igel metricq-source-sysinfo[753]:   File "/home/service/metricq-source-sysinfo/metricq_source_sysinfo/sysinfo_source.py", line 113, in update
Oct 07 10:24:34 igel metricq-source-sysinfo[753]:     prev_net_values = self.prev_net_io[nic_name]
Oct 07 10:24:34 igel metricq-source-sysinfo[753]: KeyError: 'docker0'
```

**Comment by bmario** [comment]
"Fixed" in metricq/metricq-python#135

I don't think it makes sense to handle this specific error any differently than the new default of just stopping the source (and then restarting). The internal state got messed up. Restart is a quick fix.

**Comment by bmario** [comment]
Did we actually deploy this fix? Happend again yesterday:
```
Feb 01 16:03:06 igel metricq-source-sysinfo[770]: 2023-02-01 16:03:06,603 [ERROR   ] [metricq.agent       ] Exception in event loop: Task exception was never retrieved
Feb 01 16:03:06 igel metricq-source-sysinfo[770]: 2023-02-01 16:03:06,756 [ERROR   ] [metricq.agent       ] Future: <Task finished name='Task-35' coro=<IntervalSource.task() done, defined
Feb 01 16:03:06 igel metricq-source-sysinfo[770]: 2023-02-01 16:03:06,862 [ERROR   ] [metricq.agent       ] Agent SysinfoSource encountered an unhandled exception
Feb 01 16:03:06 igel metricq-source-sysinfo[770]: Traceback (most recent call last):
Feb 01 16:03:06 igel metricq-source-sysinfo[770]:   File "/home/service/envs/sysinfo/lib/python3.9/site-packages/metricq/interval_source.py", line 131, in task
Feb 01 16:03:06 igel metricq-source-sysinfo[770]:     await self.update()
Feb 01 16:03:06 igel metricq-source-sysinfo[770]:   File "/home/service/metricq-source-sysinfo/metricq_source_sysinfo/sysinfo_source.py", line 113, in update
Feb 01 16:03:06 igel metricq-source-sysinfo[770]:     prev_net_values = self.prev_net_io[nic_name]
Feb 01 16:03:06 igel metricq-source-sysinfo[770]: KeyError: 'vethe333be2'
